pkg/logs/decoder/decoder.go

```diff
@@ -48,14 +47,20 @@ func NewOutput(content []byte, status string, rawDataLen int, timestamp string)
 type Decoder struct {
 	InputChan  chan *Input
 	OutputChan chan *Output
-
+	EndLineStrategy
```

Comment:
I struggled finding a name to isolate the two different logic to send logs either in streaming or batching mode and I came up with the name of the design pattern as well which is I think is a bad pattern.
I think we should probably rename this interface `EndlineMatcher` which provides a `MatchesEndline` method